test/base/test_workflow.py

```diff
@@ -0,0 +1,123 @@
+"""Test a basic workflow, with minimum dependencies.
+
+These tests are not for correctness, but for basic functionality.
+"""
+
+import pypesto
+import pypesto.optimize as optimize
+import pypesto.profile as profile
+import pypesto.sample as sample
+import pypesto.visualize as visualize
+
+from ..util import CRProblem
+
+
+def test_objective():
+    """Test a simple objective function."""
+    crproblem = CRProblem()
+    obj = crproblem.get_objective()
+    p = crproblem.p_true
+
+    assert obj(p) == crproblem.get_fnllh()(p)
+    assert obj(p, sensi_orders=(0,)) == crproblem.get_fnllh()(p)
+    assert (obj(p, sensi_orders=(1,)) == crproblem.get_fsnllh()(p)).all()
+    assert (obj(p, sensi_orders=(2,)) == crproblem.get_fs2nllh()(p)).all()
+    fval, grad = obj(p, sensi_orders=(0, 1))
+    assert fval == crproblem.get_fnllh()(p)
+    assert (grad == crproblem.get_fsnllh()(p)).all()
+
+
+def test_optimize():
+    """Test a simple multi-start optimization."""
+    crproblem = CRProblem()
+    problem = pypesto.Problem(
+        objective=crproblem.get_objective(),
+        lb=crproblem.lb,
+        ub=crproblem.ub,
+    )
+    optimizer = optimize.ScipyOptimizer()
+    n_start = 20
+    result = optimize.minimize(
+        problem=problem,
+        optimizer=optimizer,
+        n_starts=n_start,
+    )
+
+    # check basic sanity
+    assert len(result.optimize_result.list) == n_start
+    assert len(result.optimize_result.fval) == n_start
+    assert len(result.optimize_result.x) == n_start
+
+    # check that the results are sorted
+    fvals = result.optimize_result.fval
+    assert fvals == sorted(fvals)
+
+    # check that optimization was successful
+    assert fvals[0] < crproblem.get_fnllh()(crproblem.p_true)
+
+    # visualize the results
+    visualize.waterfall(result)
```

Comment:
use [`close_fig`](https://github.com/ICB-DCM/pyPESTO/blob/f29c848a1806c4e0fb10025978cf66bc9f9866bb/test/visualize/test_visualize.py#L27) for all tests involving plotting?